File: VB/Controllers/HomeController.vb
Imports System
Imports System.Web.Mvc
Imports Sample.Models

Namespace GridViewLinkColumnMvc.Controllers
    Public Class HomeController
        Inherits Controller
        Private personsList As New PersonsList()

        Public Function Index() As ActionResult
            Return View()
        End Function

        Public Function GridViewPartial() As ActionResult
            Return PartialView(personsList.GetPersons())
        End Function

        Public Function Details(ByVal id As Object) As ActionResult
            Return View(id)
        End Function
    End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System;
using System.Web.Mvc;
using Sample.Models;

namespace GridViewLinkColumnMvc.Controllers {
    public class HomeController : Controller {
        PersonsList personsList = new PersonsList();
        
        public ActionResult Index() {
            return View();
        }

        public ActionResult GridViewPartial() {
            return PartialView(personsList.GetPersons());
        }

        public ActionResult Details(object id) {
            return View(id);
        }
    }
}